src/newHome/homeOnboarding/onboardingContext/useOnboardingStorage.ts

```diff
@@ -0,0 +1,83 @@
+import { useUser } from "@dashboard/auth";
+import { MetadataInput, useUpdateUserMetadataMutation } from "@dashboard/graphql";
+import {
+  OnboardingState,
+  StorageService,
+} from "@dashboard/newHome/homeOnboarding/onboardingContext/types";
+import debounce from "lodash/debounce";
+import { useCallback, useMemo } from "react";
+
+const METADATA_KEY = "onboarding";
+
+export const useOnboardingStorage = (): StorageService => {
+  const { user } = useUser();
+  const [updateMetadata] = useUpdateUserMetadataMutation({});
+
+  const getOnboardingState: StorageService["getOnboardingState"] = () => {
+    try {
+      const metadata = user?.metadata.find(m => m.key === METADATA_KEY);
+
+      if (!metadata) {
+        return undefined;
+      }
+
+      return JSON.parse(metadata.value);
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.warn("Could not get onboarding state from metadata", { error });
+
+      return undefined;
```

Comment:
what happens if this occur? do you have some default state?

Reply:
nothing happened as state is kept in react state hook so the user refresh the page state will be restarted to last one saved in metadata or initial one when never was saved in metadata